### PR TITLE
Make loggest a Windows service

### PR DESCRIPTION
**Pull request by gleon99**

### CODE REVIEW

File: .github/workflows/build-msi.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      - 'loggestd/*'
+
+jobs:
+  build-loggestd-msi:
+    runs-on: windows-latest
+    steps:
+      - uses: hecrj/setup-rust-action@master
+      - uses: actions/checkout@master
+      - name: Cargo build
+        run: cargo build --all --release
```
Comment (drrlvn) [inline .github/workflows/build-msi.yml]: Don't need `--all`, you only build one crate and not all the workspace.

File: .github/workflows/build-msi.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        working-directory: loggestd
+      - name: Cargo test
+        run: cargo test --all
```
Comment (drrlvn) [inline .github/workflows/build-msi.yml]: Same think with the `--all`.

File: .github/workflows/build-msi.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        run: '"%PROGRAMFILES(x86)%\WiX Toolset v3.11\bin\candle.exe" "loggestd.wxs"'
+        working-directory: loggestd
+      - name: Link MSI
+        run: '"%PROGRAMFILES(x86)%\WiX Toolset v3.11\bin\light.exe" "loggestd.wixobj"'
+        working-directory: loggestd
+      - name: Release
+        uses: softprops/action-gh-release@v1
+        if: startsWith(github.ref, 'refs/tags/') && contains(github.ref, 'loggestd')
+        with:
+          files: loggestd.msi
```
Comment (drrlvn) [inline .github/workflows/build-msi.yml]: Filename should contain the version.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        os: [ubuntu-16.04, macOS-latest, windows-latest]
+        directory: [., loggestd, ioym]
+
+    steps:
+      - uses: hecrj/setup-rust-action@master
+      - uses: actions/checkout@master
+      - name: Install rustfmt
+        run: rustup component add rustfmt
+      - name: rustfmt
+        run: cargo fmt --all -- --check
```
Comment (drrlvn) [inline .github/workflows/ci.yml]: Either remove `--all` or the `working-directory`.

File: loggestd/src/log_file.rs [hunk 5]
```diff
@@ -23,34 +23,12 @@ fn generate_filename(base_name: &Path, index: usize) -> PathBuf {
...
-    ensure_directory(directory)
-}
-
 impl LogFile {
     pub fn open(base_filename: PathBuf) -> Result<Self, io::Error> {
         let index = 1;
         let filename = generate_filename(&base_filename, index);
 
-        recursive_ensure_directory(filename.parent().unwrap())?;
+        create_dir_all(filename.parent().unwrap())?;
```
Comment (drrlvn) [inline loggestd/src/log_file.rs]: This change should be in a separate commit, possibly a separate PR.

File: loggestd/src/main.rs [hunk 6]
```diff
@@ -1,26 +1,60 @@
...
 mod usage_monitor;
 
-fn main() {
+#[cfg(windows)]
+const SERVICE_NAME: &str = "Loggest";
+#[cfg(windows)]
+const SERVICE_TYPE: service::ServiceType = service::ServiceType::OWN_PROCESS;
+
+#[cfg(windows)]
+fn wait(recv: crossbeam_channel::Receiver<()>) -> impl Future<Item = (), Error = ()> {
```
Comment (drrlvn) [inline loggestd/src/main.rs]: Probably should have a better name.

**Comment by drrlvn** [comment]
Also, since you migrated the CI to GitHub Actions please remove Azure Pipelines.

File: loggestd/src/log_file.rs (hunk 5 above)
Comment (gleon99) [reply loggestd/src/log_file.rs]: Done
